File: backend/app/schemas/customer.py
from datetime import datetime

from pydantic import (
    BaseModel,
    ConfigDict,
    Field,
)


class CustomerUpdate(BaseModel):

    company_name: str | None = Field(
        default=None,
        min_length=1,
        max_length=200,
    )

    contact_person: str | None = Field(
        default=None,
        max_length=150,
    )

    email: str | None = Field(
        default=None,
        max_length=150,
    )

    phone: str | None = Field(
        default=None,
        max_length=30,
    )

    address: str | None = Field(
        default=None,
        max_length=500,
    )

    city: str | None = Field(
        default=None,
        max_length=100,
    )

    state: str | None = Field(
        default=None,
        max_length=100,
    )

    pincode: str | None = Field(
        default=None,
        max_length=20,
    )


class CustomerResponse(BaseModel):

    id: int

    customer_code: str

    company_name: str

    contact_person: str | None = None

    email: str | None = None

    phone: str | None = None

    gst_number: str | None = None

    address: str | None = None

    city: str | None = None

    state: str | None = None

    pincode: str | None = None

    # Legacy/internal field.
    # Not exposed as a business control in the frontend.
    is_active: bool

    created_at: datetime

    updated_at: datetime | None = None

    model_config = ConfigDict(
        from_attributes=True
    )

File: frontend/src/types/customer.ts
export interface Customer {
  id: number;

  customer_code: string;

  company_name: string;

  contact_person: string | null;

  email: string | null;

  phone: string | null;

  gst_number: string | null;

  address: string | null;

  city: string | null;

  state: string | null;

  pincode: string | null;

  // Internal legacy field only.
  // Not controlled from the Customer UI.
  is_active: boolean;

  created_at: string;

  updated_at: string | null;
}


export interface CustomerUpdatePayload {
  company_name: string;

  contact_person: string;

  email: string;

  phone: string;

  address: string;

  city: string;

  state: string;

  pincode: string;
}